crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -114,8 +114,10 @@ impl RoomList {
     }
 
     /// Get a subscriber to the room list loading state.
+    ///
+    /// This method will send out the current loading state as the first update.
     pub fn loading_state(&self) -> Subscriber<RoomListLoadingState> {
-        self.loading_state.subscribe()
+        self.loading_state.subscribe_reset()
```

Comment:
Why do you need this?

Reply:
Without that a complement-crypto test is flaky because the test is waiting on a `Loading` event which can not come until the incremental sync timeouts, if there is no new data.

You can read the unthreaded comments with Keygan investigation around [here](https://github.com/matrix-org/matrix-rust-sdk/pull/4035#issuecomment-2385437240).